elastalert/elastalert.py

```diff
@@ -1986,7 +1987,8 @@ def get_top_counts(self, rule, starttime, endtime, keys, number=None, qk=None):
                 buckets = list(hits_terms.values())[0]
 
                 # get_hits_terms adds to num_hits, but we don't want to count these
-                self.thread_data.num_hits -= len(buckets)
+                if getattr(self.thread_data, "num_hits", None):
```

Comment:
This fixes a bug I met when having `top_count_keys` in a frequency rule with `aggregation`. 

"num_hits" was undefined.